test/e2e/v1beta1/scripts/gh-actions/run-e2e-tune-api.py

```diff
@@ -11,8 +12,68 @@
 # The default logging config.
 logging.basicConfig(level=logging.INFO)
 
+def run_e2e_experiment_create_by_tune_custom_metrics_collector(
```

Comment:
I don't think we need another E2E just to test this functionality.
We already run E2Es using these YAML files for various metrics collectors: https://github.com/kubeflow/katib/tree/master/examples/v1beta1/metrics-collector
I think for your feature, you just need to add unit tests for Katib Client to verify that Experiment has correct specification: https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/api/katib_client_test.py

Reply:
Sure! should I then remove the e2e test with custom metric collector from `run-e2e-tune-api.py`?

Reply:
Yes, please can you add the unit tests for your functionality: https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/api/katib_client_test.py ?